zoobot/pytorch/training/finetune.py

```diff
@@ -104,37 +101,45 @@ def __init__(
     ):
         super().__init__()
 
         # adds every __init__ arg to model.hparams
         # will also add to wandb if using logging=wandb, I think
         # necessary if you want to reload!
         # with warnings.catch_warnings():
-            # warnings.simplefilter("ignore")
-            # this raises a warning that encoder is already a Module hence saved in checkpoint hence no need to save as hparam
-            # true - except we need it to instantiate this class, so it's really handy to have saved as well
-            # therefore ignore the warning
-        self.save_hyperparameters(ignore=['encoder']) # never serialise the encoder, way too heavy
-            # if you need the encoder to recreate, pass when loading checkpoint e.g. 
-            # FinetuneableZoobotTree.load_from_checkpoint(loc, encoder=encoder)
-        
+        # warnings.simplefilter("ignore")
+        # this raises a warning that encoder is already a Module hence saved in checkpoint hence no need to save as hparam
+        # true - except we need it to instantiate this class, so it's really handy to have saved as well
+        # therefore ignore the warning
+        self.save_hyperparameters(ignore=["encoder"])  # never serialise the encoder, way too heavy
+        # if you need the encoder to recreate, pass when loading checkpoint e.g.
+        # FinetuneableZoobotTree.load_from_checkpoint(loc, encoder=encoder)
+
         if name is not None:
-            assert encoder is None, 'Cannot pass both name and encoder to use'
+            assert encoder is None, "Cannot pass both name and encoder to use"
             self.encoder = timm.create_model(name, num_classes=0, pretrained=True)
             self.encoder_dim = self.encoder.num_features
 
         elif zoobot_checkpoint_loc is not None:
-            assert encoder is None, 'Cannot pass both checkpoint to load and encoder to use'
-            self.encoder = load_pretrained_zoobot(zoobot_checkpoint_loc)  # extracts the timm encoder
+            assert encoder is None, "Cannot pass both checkpoint to load and encoder to use"
+            self.encoder = load_pretrained_zoobot(
+                zoobot_checkpoint_loc
+            )  # extracts the timm encoder
             self.encoder_dim = self.encoder.num_features
+
         else:
-            assert zoobot_checkpoint_loc is None, 'Cannot pass both checkpoint to load and encoder to use'
-            assert encoder is not None, 'Must pass either checkpoint to load or encoder to use'
+            assert (
+                zoobot_checkpoint_loc is None
+            ), "Cannot pass both checkpoint to load and encoder to use"
+            assert encoder is not None, "Must pass either checkpoint to load or encoder to use"
             self.encoder = encoder
             # work out encoder dim 'manually'
-            self.encoder_dim = define_model.get_encoder_dim(self.encoder)
+            if isinstance(self.encoder, timm.models.VisionTransformer):
```

Comment:
TODO for me, I later found out this actually works on all timm models so should add a check like this for everything

Reply:
Changed to read attrs

Reply:
Changed to read attrs